ceph_iscsi_config/gateway_setting.py

```diff
@@ -0,0 +1,185 @@
+
+    if isinstance(value, bool):
+        return value
+
+    value = str(value).lower()
+    if value in ['1', 'true', 'yes']:
+        return True
+    elif value in ['0', 'false', 'no']:
+        return False
+    raise ValueError(value)
+
+
+class Setting(object):
+    def __init__(self, name, type_str, def_val):
+        self.name = name
+        self.type_str = type_str
+        self.def_val = def_val
+
+    def __contains__(self, key):
+        return key == self.def_val
+
+
+class BoolSetting(Setting):
+    def __init__(self, name, def_val):
+        super(BoolSetting, self).__init__(name, "bool", def_val)
+
+    def to_str(self, norm_val):
+        if norm_val:
+            return "true"
+        else:
+            return "false"
+
+    def normalize(self, raw_val):
+        try:
+            # for compat we also support Yes/No and 1/0
+            return convert_str_to_bool(raw_val)
+        except ValueError:
+            raise ValueError("expected true or false for {}".format(self.name))
+
+
+class LIOBoolSetting(BoolSetting):
+    def __init__(self, name, def_val):
+        super(LIOBoolSetting, self).__init__(name, def_val)
+
+    def to_str(self, norm_val):
+        if norm_val:
+            return "yes"
+        else:
+            return "no"
+
+    def normalize(self, raw_val):
+        try:
+            # for compat we also support True/False and 1/0
+            return convert_str_to_bool(raw_val)
+        except ValueError:
+            raise ValueError("expected yes or no for {}".format(self.name))
+
+
+class ListSetting(Setting):
+    def __init__(self, name, def_val):
+        super(ListSetting, self).__init__(name, "list", def_val)
+
+    def to_str(self, norm_val):
+        return str(norm_val)
+
+    def normalize(self, raw_val):
+        return raw_val.split(',') if raw_val else []
+
+
+class StrSetting(Setting):
+    def __init__(self, name, def_val):
+        super(StrSetting, self).__init__(name, "str", def_val)
+
+    def to_str(self, norm_val):
+        return str(norm_val)
+
+    def normalize(self, raw_val):
+        return str(raw_val)
+
+
+class IntSetting(Setting):
+    def __init__(self, name, min_val, max_val, def_val):
+        self.min_val = min_val
+        self.max_val = max_val
+        super(IntSetting, self).__init__(name, "int", def_val)
+
+    def to_str(self, norm_val):
+        return str(norm_val)
+
+    def normalize(self, raw_val):
+        try:
+            val = int(raw_val)
+        except ValueError:
+            raise ValueError("expected integer for {}".format(self.name))
+
+        if val < self.min_val:
+            raise ValueError("expected integer >= {} for {}".
+                             format(self.min_val, self.name))
+        if val > self.max_val:
+            raise ValueError("expected integer <= {} for {}".
+                             format(self.max_val, self.name))
+        return val
+
+
+class EnumSetting(Setting):
+    def __init__(self, name, def_val, valid_vals):
```

Comment:
Nit: I've noticed that `def_val` is the last arg in `IntSetting` and the first in the `EnumSetting`, does it make sense to uniformize for better readability?

This is what we have now:
```
class IntSetting(Setting):
    def __init__(self, name, min_val, max_val, def_val):
class EnumSetting(Setting):
    def __init__(self, name, def_val, valid_vals):
```
I was expecting something like:
```
class IntSetting(Setting):
    def __init__(self, name, def_val, min_val, max_val):
class EnumSetting(Setting):
    def __init__(self, name, def_val, valid_vals):
```
or 
```
class IntSetting(Setting):
    def __init__(self, name, min_val, max_val, def_val):
class EnumSetting(Setting):
    def __init__(self, name, valid_vals, def_val):
```

Reply:
That is a bug.